### PR TITLE
Coin change problem

### PR DESCRIPTION
**Pull request by nandikeshsingh**
## ISSUE NUMBER

#241 

## SHORT DESCRIPTION

Coin change problem using dynamic programming in C++.

## TESTING

Passed all the test cases on geeksforgeeks.
screenshot from 2017-12-18 15-49-58

### CODE REVIEW

File: Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Problem :-
+Given a value N and the values of different denomination,find the number of ways to make changes for N cents,if we have infinite supply of all denominations.
```
Comment (prateekiiest) [inline Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md]: make problem,input, output and others alike as a heading (### )

File: Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Input : Size of an array and all the values in the array and the number of cents.
+
+Output : An integer that denotes the  number of ways to make change.
+
+Language : C++
+
+Algorithm Paradigm : Dynamic Programming
+
+Time Complexity : O(N*M)
```
Comment (prateekiiest) [inline Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md]: make language, algorithm paradigm and time complexity bold.

use tilde tag `O(N*M)`

**Review by prateekiiest — CHANGES_REQUESTED** [review]

**Comment by nandikeshsingh** [comment]
updated the readme as needed.

**Comment by nandikeshsingh** [comment]
changes done as required.

**Comment by nandikeshsingh** [comment]
updated readme as needed.

**Comment by nandikeshsingh** [comment]
updated readme as needed.

File: Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Input:-
+---
+Size of an array and all the values in the array and the number of cents.
+
+Output :-
+---
+An integer that denotes the  number of ways to make change.
+
+Language : `C++`
```
Comment (prateekiiest) [inline Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md]: don't make language, algo paradigm till space complexity a heading.

use #### to make those a subheading

File: Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+Algorithm Paradigm : `Dynamic Programming`
+---
+Time Complexity :  `O(N*M)`
+---
+Space Complexity : `O(N*M)`
+---
+Working :-
+---
+An arr array stores the different currency values.
+Another array temp stores the best values for sub-problems.For eg: temp[3][4] stores the optimal value for the number of ways
```
Comment (prateekiiest) [inline Competitive Coding/Dynamic Programming/Coin Change/Coin Change Readme.md]: use `temp[3][4]` before and after using tilde (`).

Similarly for arr[0]., arr[1], ... use tilde `

**Review by prateekiiest — CHANGES_REQUESTED** [review]

**Comment by nandikeshsingh** [comment]
updated readme.md

**Review by prateekiiest — APPROVED** [review]

**Pull request merged**